orchagent/pfcactionhandler.cpp

```diff
@@ -533,6 +535,22 @@ PfcWdZeroBufferHandler::~PfcWdZeroBufferHandler(void)
         SWSS_LOG_ERROR("Failed to set buffer profile ID on queue 0x%" PRIx64 ": %d", getQueue(), status);
         return;
     }
+
+    lockUnlockPriorityGroupAndQueue(portInstance, false);
+}
+
+void PfcWdZeroBufferHandler::lockUnlockPriorityGroupAndQueue(Port& port, bool lock) const
+{
+    // set lock bits on PG and queue
+    port.m_priority_group_lock[static_cast<size_t>(getQueueId())] = lock;
+    for (size_t i = 0; i < port.m_queue_ids.size(); ++i)
+    {
+        if (port.m_queue_ids[i] == getQueue())
+        {
+            port.m_queue_lock[i] = lock;
+        }
+    }
+    gPortsOrch->setPort(port.m_alias, port);
```

Comment:
>gPortsOrch->setPort(port.m_alias, port); [](start = 4, length = 40)

Is it really needed? I think port is already in gPortsOrch->m_portList.

Reply:
It is needed, gPortsOrch->getPort returns a copy of Port